zulipterminal/api_types.py

```diff
@@ -27,7 +27,8 @@ class Message(TypedDict, total=False):
     id: int
     sender_id: int
     content: str
-    recipient_id: int
+    # NOTE: Deprecated; a server implementation detail not useful in a client.
+    # recipient_id: int
```

Comment:
I think it would be reasonable to move this to the bottom of the class into an unsupported/unused section of comments, with details on the status of each (ie. when they were removed/unused/deprecated/etc).

As you note, `recipient_id` is only deprecated and not removed, at least for now.

However, `sender_short_name` was removed in Zulip 3.1 (check the api changelog) - and we haven't used it or similar short names - so it would be good to do other tidying along these lines too.